### ISSUE TITLE
"open /storage/: no such file or directory" - Docker on Windows

### ISSUE THREAD

**Issue by Tohaker** [fsouza/fake-gcs-server#114]
I'm trying to run the server, here is my command to start the Docker container:
`docker run -d --name fake-gcs-server -p 4443:4443 -v %CD%\bot\tests\data:/data fsouza/fake-gcs-server`

It starts with no problems, but using both cURL and Postman gives the same response when hitting `https://127.0.0.1:4443/storage/v1/b`:

`open /storage/: no such file or directory`

I'm clearly missing something here, but can't think what...

**Comment by fsouza** [comment]
Hey @Tohaker, thanks for opening this issue. This is weird 😞 Do you see anything in the container logs or more information in the response data?

Sorry for the delayed response x)

**Comment by Tohaker** [comment]
Here are the logs:

**Comment by Tohaker** [comment]
As suspected, I flattened the folder structure and it worked. Is there any way to make this work with nested folders? My real bucket has a directory structure that I'd like to test against.

**Comment by fsouza** [comment]
@Tohaker gotcha, thanks for posting the logs. Yeah, currently we don't support recursively creating the objects, but we could implement it, checking for a file/directory here: https://github.com/fsouza/fake-gcs-server/blob/10d4e3a3d24667ab71ffd590c10f3046b6224e8c/main.go#L69

I may be able to do it over the weekend depending on how cold it gets 😁

**Comment by Tohaker** [comment]
That would be much appreciated 😄 Sadly I'm not familiar with GO otherwise I'd try to submit a PR myself.

**Comment by fsouza** [comment]
@Tohaker no worries, I got you

**Comment by gerardmrk** [comment]
Same issue here, using Docker Desktop for MacOS. Part of the compose file:

```
...

blobstore:
    container_name: blobstore
    build:
      context: ./blobstore
      dockerfile: Dockerfile
    ports:
      - "4443:4443/tcp"
    volumes:
      - "/usr/local/var/blobstore:/data"

...
```

Running the example curls:

```bash
$ open /storage/: no such file or directory
```

Running docker logs:

```bash
$ time="2019-12-09T00:59:42Z" level=info msg="couldn't load any objects or buckets from \"/data\", starting empty"
$ time="2019-12-09T00:59:42Z" level=info msg="server started at https://[::]:4443"
```

If this is useful info:
- No issues with the volume mounting; it gets created on host at the specified path.
- Volume directory stays empty before and after running the failed command (maybe that's normal).
- OS: `MacOS Catalina`, shell: `zsh`.

**Comment by gerardmrk** [comment]
sorry ignore this. It was a typo in the copy-paste

**Comment by gerardmrk** [comment]
Can confirm that mapping volume `/tmp:/data` works. My previous volume mapping `/usr/local/var/blobstore:/data`

**Comment by kaime** [comment]
I'm having the same issue when mounting a data volume that's initially empty:

``` bash
$ mkdir data
$ docker run --rm -p 4443:4443 -v ${PWD}/data:/data fsouza/fake-gcs-server  
time="2019-12-21T19:26:04Z" level=info msg="couldn't load any objects or buckets from \"/data\", starting empty"
time="2019-12-21T19:26:04Z" level=info msg="server started at https://[::]:4443"
```

``` bash
$ curl -i --insecure https://localhost:4443/storage/v1/b
HTTP/1.1 500 Internal Server Error
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Sat, 21 Dec 2019 19:31:05 GMT
Content-Length: 42

open /storage/: no such file or directory
```
Adding a trailing slash to the URL generates a different error:

``` bash
$ curl -i --insecure https://localhost:4443/storage/v1/b/
HTTP/1.1 404 Not Found
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Sat, 21 Dec 2019 19:31:38 GMT
Content-Length: 19

404 page not found
```

**Comment by gerardmrk** [comment]
@kaime try the technique mentioned above, don't do nested paths (unfortunately). just do `/tmp/data` or something more specific than `data` to avoid potential name conflicts in `/tmp`.

Alternatively, if you're using Go, checkout gocloud.dev, they provide seamless abstraction over your object storage solution, using either in-memory or local filesystem for local development.

**Comment by fsouza** [comment]
@gerardmrk sorry, seeing this now. Will reopen the issue so it's easier to track.

Is ``/usr/local/var/blobstore`` an empty directory?

**Comment by fsouza** [comment]
Oh ok I see what the issue is. Just pushed a fix and will tag 1.16.3 shortly.
